packages/web/components/funnels/concentrated-liquidity/use-cfmm-to-cl-migration.ts

```diff
@@ -23,66 +27,103 @@ export function useCfmmToClMigration(cfmmPoolId: string): {
     derivedDataStore,
   } = useStore();
 
-  const osmosisAccount = accountStore.getWallet(chainId)?.osmosis;
+  const account = accountStore.getWallet(chainId);
+  const osmosisAccount = account?.osmosis;
   const osmosisQueries = queriesStore.get(chainId).osmosis!;
-  const { sharePoolDetail } = derivedDataStore.getForPool(cfmmPoolId);
+
+  const userPoolIds = osmosisQueries.queryGammPoolShare.getOwnPools(
+    account?.address ?? ""
+  );
+
+  const firstClLinkedUserPoolId = userPoolIds.find((poolId) =>
+    Boolean(
+      osmosisQueries.queryCfmmToConcentratedLiquidityPoolLinks.get(poolId)
+        .concentratedLiquidityPoolId
+    )
+  );
+
+  const cfmmPoolId_ = cfmmPoolId ?? firstClLinkedUserPoolId ?? "";
```

Comment:
What do you think about renaming the param to cfmmPoolIdParam and remove _ from the variable.